alarm-frontend/src/Components/Alarm.js

```diff
@@ -1,63 +1,67 @@
+    const [alarm] = useState(new Audio(Sound))
+    const { realHour, realMinute, amPm } = useContext(Context);
+    const [alarmTime, setAlarmTime] = useState();
+    const [alarmExists, setAlarmExists] = useState("");
 
     const setAlarm = () => {
+        console.log("You have set an alarm!")
         setAlarmExists(true);
-        setAlarmTime(`${hour}${minutes}${amPmOption}`)
+        setAlarmTime(`${hour}${minute}${AMorPM}`);
     };
 
-    const stopAlarm = () => {
-            pauseAlarm();
-            setAlarmExists(false);
+    const noAlarm = () => {
+        console.log("Alarm is turned off!")
+        setAlarmExists(false);
+        setAlarmTime("");
+    }
+
+    const pauseAlarm = () => {
+        console.log("Alarm is paused")
+        alarm.pause();
+    }
+
+    async function deleteAlarm() {
+        await axios.post(`${BASE_URL}/alarms/${id}/delete`);
+        handler();
+    }
+
+    if (hour < 10) hour = `0${hour}`
+    if (minute < 10) minute = `0${minute}`
+
+    if (alarmTime === `${realHour}${realMinute}${amPm}`) {
+        console.log("Alarm is going off!");
+        alarm.play();
+        alarm.loop = true;
     }
 
+    useEffect(() => {
+        console.log("useEffect set an alarm")
+        setAlarmExists(true);
+        setAlarmTime(`${hour}${minute}${AMorPM}`);
+      }, [hour, minute, AMorPM]);
+
     return (
         <div>
-            <div>
-                <select {...setHour}>
-                    <option disabled value="Hour">
-                        Hour
-                    </option>
-                    {hourNumber.map((hour, index) => (
-                    <option key={index} value={hour}>
-                            {hour}    
-                    </option>
-                    ))}
-                </select>
-                <select {...setMinutes}>
-                    <option disabled value="Minutes">
-                        Minutes
-                    </option>
-                    {minutesNumber.map((minutes, index) => (
-                    <option key={index} value={minutes}>
-                            {minutes}    
-                    </option>
-                    ))}
-                </select>
-                <select {...setAmPmOption}>
-                    <option disabled value="Am-Pm">
-                        Am/Pm
-                    </option>
-                    <option value="AM">Am</option>
-                    <option value="PM">Pm</option>
-                </select>
-            </div>
-            <button onClick={alarmExists ? stopAlarm : setAlarm}>
-                        {alarmExists ? "Clear Alarm" : "Set Alarm"}
-            </button>
+            <h2>{`${hour}:${minute} ${AMorPM}`}</h2>
+            <p>{`Type: ${type}`}</p>
+            <p>{`Difficulty: ${difficulty}`}</p>
+
+            <label className="switch">
+                <input type="checkbox" onClick={alarmExists ? noAlarm : setAlarm} defaultChecked={true}></input>
```

Comment:
dont think you need the = true here
```suggestion
                <input type="checkbox" onClick={alarmExists ? noAlarm : setAlarm} defaultChecked />
```